src/Detail/Importing/Source/DBALSource.php

```diff
@@ -2,28 +2,105 @@
 
 namespace Detail\Importing\Source;
 
-use Doctrine\DBAL;
+use Doctrine\DBAL\Driver as DBALDriver;
 
 abstract class DBALSource extends BaseSource
 {
     /**
-     * @var DBAL\Connection $connection
+     * @var DBALDriver\Connection $connection
      */
     protected $connection;
 
     /**
-     * @param DBAL\Connection $connection
+     * @param DBALDriver\Connection $connection
      */
-    public function __construct(DBAL\Connection $connection)
+    public function __construct(DBALDriver\Connection $connection)
     {
         $this->connection = $connection;
     }
 
     /**
-     * @return DBAL\Connection
+     * @return array
+     */
+    public function fetchAll()
+    {
+        return $this->findBy();
+    }
+
+    /**
+     * @return DBALDriver\Connection
      */
     protected function getConnection()
     {
         return $this->connection;
     }
+
+    /**
+     * @return string
+     */
+    abstract protected function getSelectString();
+
+    /**
+     * @param array $criteria
+     * @return array
+     */
+    protected function findBy(
+        array $criteria = array()
+//        array $orderBy = null,
+//        $limit = null,
+//        $offset = null
+    ) {
+        $statement = $this->createSelectQuery($criteria);
+
+        if (!$statement->execute()) {
+            return array();
+        };
+
+        return $statement->fetchAll(\PDO::FETCH_ASSOC);
+    }
+
+    /**
+     * @param array $criteria
+     * @return DBALDriver\Statement
+     */
+    protected function createSelectQuery(
```

Comment:
This code is wrong here, should be defined as abstract method and implemented in the extending class (Code will be specific for the Adapter Layer e.g. MsSql)